README.md

```diff
@@ -1,7 +1,52 @@
 # C++ project template
 
-# Setup
+## Template Initialization
 
-```bash
+This project is a template that cannot be used before initialization. To initialize your project, you must run the following shell command:
+
+```sh
 cmake -P init.cmake --project <name> --module <name> --header <name>
 ```
+
+- **Project name** will become your top-level CMake project name
+  - Must be alphanumeric and begin with a character.
+- **Module name** will be concatenated with your project name to determine the library's namespace (`${project}::${module}`)
+- **Header name** will become the name of your library's main include header
+
+
+## Building
+
+To build the project locally, you will need to select a [CMake](https://cmake.org/) preset that matches your system configuration. Your system's configuration is described by a [triplet](https://wiki.osdev.org/Target_Triplet).
+
+Presets for the most common system triplets are defined in [`CMakePresets.json`](./CMakePresets.json). 
+
+Notice that each system triplet defines a preset for multiple compilers. If you have a compiler preference, you can pick the respective preset. If you do not have a preference, you can choose [GCC](https://gcc.gnu.org/), which should be available on most systems.
+
+Now you can configure your project:
+
+```sh
+cmake --preset=<PRESET>
```

Comment:
`cmake --workflow --preset <name>`
that will run:
-  configure step - cmake --preset <name>
-  build step        -  cmake --build --preset <name>
-  test step          -  ctest --preset <name>

Each step can be run manually.